src/Route.php

```diff
@@ -36,13 +37,23 @@ public function __construct(
         $this->url = '/' . trim($url, '/');
         $this->metadata = new Metadata($metadata);
 
+        if (str_contains($this->url, '*')) {
+            if (!str_ends_with($this->url, '*')) {
+                throw new RoutingException('Catch-all parameter * can only be at the end of the route URL.');
+            }
+            if ($this->url !== '*' && !str_ends_with($this->url, '/*')) {
```

Comment:
This check does not seem to check what the exception says. What it's checking is that the url isn't JUST `*` and doesn't end with `*`, if I'm seeing this correctly. It's still possible for it to contain multiple asterisks.

Reply:
You're right, quite the overlook...